go/cmd/sqlflowserver/e2e_common_cases.go

```diff
@@ -669,78 +685,94 @@ func caseXGBoostSparseKeyValueColumn(t *testing.T) {
 		return columns
 	}
 
-	dropDBSQL := ""
-	if testDBType == "hive" {
-		// Hive can only drop non-empty database in the CASCADE mode.
-		dropDBSQL = fmt.Sprintf("DROP DATABASE IF EXISTS %s CASCADE;", dbName)
-	} else {
-		dropDBSQL = fmt.Sprintf("DROP DATABASE IF EXISTS %s;", dbName)
+	if !isMaxCompute {
+		defer executeSQLFunc(dropDBSQL)
 	}
 
-	defer executeSQLFunc(dropDBSQL)
-
-	prepareDataSQL := []string{
-		dropDBSQL,
-		fmt.Sprintf(`CREATE DATABASE IF NOT EXISTS %s;`, dbName),
-		fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s.train (c1 VARCHAR(255), label_col FLOAT);`, dbName),
-		fmt.Sprintf(`INSERT INTO %s.train VALUES("1:0.5 3:-1.5 5:6.7", 0.5), ("3:-1.3 10:0.6 2:-3.4", -0.5);`, dbName),
+	var prepareDataSQL []string
+	if !isMaxCompute {
+		prepareDataSQL = []string{
+			dropDBSQL,
+			fmt.Sprintf(`CREATE DATABASE IF NOT EXISTS %s;`, dbName),
+			fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s.%s (c1 VARCHAR(255), label_col FLOAT);`, dbName, trainTable),
+		}
+	} else {
+		prepareDataSQL = []string{
+			fmt.Sprintf(`DROP TABLE IF EXISTS %s.%s;`, dbName, trainTable),
+			fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s.%s (c1 STRING, label_col DOUBLE);`, dbName, trainTable),
+		}
 	}
 
+	insertDataSQL := fmt.Sprintf(`INSERT INTO %s.%s VALUES("1:0.5 3:-1.5 5:6.7", 0), 
+("3:-1.3 10:0.6 2:-3.4", 1), ("5:10.2 3:7.4", 2);`, dbName, trainTable)
+	prepareDataSQL = append(prepareDataSQL, insertDataSQL)
+
 	for _, sql := range prepareDataSQL {
 		executeSQLFunc(sql)
 	}
 
-	trainedModel := fmt.Sprintf("%s.trained_model", dbName)
+	trainedModel := "xgb_kv_column_trained_model"
+	if !isMaxCompute {
+		trainedModel = fmt.Sprintf("%s.%s", dbName, trainedModel)
+	}
 
-	const trainSQLTemplate = `SELECT c1, label_col FROM %s.train
-TO TRAIN xgboost.gbtree
-WITH 
-	objective = "reg:squarederror",
-	train.num_boost_round = 20
-COLUMN SPARSE(c1%s)
-LABEL label_col
-INTO %s;
-`
-
-	trainSQL := fmt.Sprintf(trainSQLTemplate, dbName, "", trainedModel)
+	const trainSQLTemplate = `SELECT c1, label_col FROM %s.%s
+	TO TRAIN xgboost.gbtree
+	WITH
+		num_class = 3,
+		objective = "multi:softprob",
+		train.num_boost_round = 20
+	COLUMN SPARSE(c1%s)
+	LABEL label_col
+	INTO %s;
+	`
+
+	trainSQL := fmt.Sprintf(trainSQLTemplate, dbName, trainTable, "", trainedModel)
 	executeSQLFunc(trainSQL)
-	hasModelTableFunc(trainedModel)
+	if !isMaxCompute {
+		hasModelTableFunc(trainedModel)
```

Comment:
Why? model is saved to OSS only when `SQLFLOW_submitter=pai`, this test should have the model table.

Reply:
Thanks! Fixed.